src/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str_str.cs

```diff
@@ -1,221 +1,378 @@
+                        break;
+                    //We dont throw in V1 too
+                    case ':':
+                        //History:
+                        // 1) we assumed that this will work in all non-9x machine
+                        // 2) Then only in XP
+                        // 3) NTFS?
+                        if (Interop.IsWindows && FileSystemDebugInfo.IsCurrentDriveNTFS()) // testing NTFS
+                            Assert.Throws<IOException>(() => GetEntries(Directory.GetCurrentDirectory(), string.Format("te{0}st", invalidFileNames[i].ToString())));
+                        else
+                            GetEntries(Directory.GetCurrentDirectory(), string.Format("te{0}st", invalidFileNames[i].ToString()));
 
-    #endregion
-}
+                        break;
+                    case '*':
+                    case '?':
+                        GetEntries(Directory.GetCurrentDirectory(), string.Format("te{0}st", invalidFileNames[i].ToString()));
+                        break;
+                    default:
+                        Assert.Throws<ArgumentException>(() => GetEntries(Directory.GetCurrentDirectory(), string.Format("te{0}st", invalidFileNames[i].ToString())));
+                        break;
+                }
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.AnyUnix)]
+        public void UnixSearchPatternInvalid()
+        {
+            Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, "\0"));
+            Assert.Throws<ArgumentException>(() => GetEntries(TestDirectory, string.Format("te{0}st", "\0".ToString())));
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsSearchPatternQuestionMarks()
+        {
+            string testDir1Str = GetTestFileName();
+            DirectoryInfo testDir = new DirectoryInfo(TestDirectory);
+            DirectoryInfo testDir1 = testDir.CreateSubdirectory(testDir1Str);
+
+            using (File.Create(Path.Combine(TestDirectory, testDir1Str, GetTestFileName())))
+            using (File.Create(Path.Combine(TestDirectory, GetTestFileName())))
+            {
+                string[] results = GetEntries(TestDirectory, string.Format("{0}.???", new string('?', GetTestFileName().Length)));
+                if (TestFiles && TestDirectories)
+                    Assert.Equal(2, results.Length);
+                else
+                    Assert.Equal(1, results.Length);
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void WindowsSearchPatternWhitespace()
+        {
+            Assert.Empty(GetEntries(TestDirectory, "           "));
+            Assert.Empty(GetEntries(TestDirectory, "\n"));
+            Assert.Empty(GetEntries(TestDirectory, " "));
+            Assert.Empty(GetEntries(TestDirectory, "\t"));
```

Comment:
I actually just learned about the [Theory] tag about 2 minutes ago from looking at your MemoryMapped PR! Definitely going to be using it in the next tests I write; there's several occasions in this code where I could have made use of it.